src/resources/nfts/index.ts

```diff
@@ -99,3 +115,24 @@ export function useLegacyNFTs({ address }: { address: string }) {
     isInitialLoading: !nfts.length && isFetching,
   };
 }
+
+export function useNFTListing({
+  contractAddress,
+  tokenId,
+  network,
+}: {
+  contractAddress: string;
+  tokenId: string;
+  network: Omit<Network, Network.goerli>;
+}) {
+  return useQuery(
+    nftListingQueryKey({ contractAddress, tokenId, network }),
+    async () =>
+      (await fetchSimpleHashNFTListing(contractAddress, tokenId, network)) ??
+      null,
```

Comment:
Do we really want `null` instead of `undefined` here like it would return otherwise? Might be a fine application of it, since it's not NOT defined, it's just not available. 

Imo if we're just going to check for like `listing = useNFTListing()` and `if (!listing) {` then `undefined` is fine and that way we can let TypeScript just infer `fetchSimpleHashListing`'s return type.

Similar note, if we want `fetchSimpleHashListing` to return `null`, we can do that there instead of here.

Reply:
useQuery actually does not allow undefined to be returned by its query function

Reply:
I did not know that, interesting